### PR TITLE
Dockernet Linux Support

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #XXX

## Context and purpose of the change
* Up to this point, dockernet has only been tested on MacOS. This PR introduces changes to make it compatible with linux. 
* This was tested using a Debian VM in GCP
* Shoutout @nghuyenthevinh2000 for the majority of these changes!

## Brief Changelog
* Updated submodules to use https instead of ssh
* Changed shell from `sh` to `bash`
* Fixed bug in go build command for host binaries
* Added dockernet Stride Admin address to vars
* Disabled hermes to allow easier startup (hermes takes the longest to build and requires the most dependencies)

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

**Comment by faddat** [comment]
amusingly, I created my own variant of this too.

**Comment by faddat** [comment]
still fails on my machine.  

Suggest we get rid of all reference to brew?
image

**Comment by faddat** [comment]
....and let's try and add this to ci....

**Comment by sampocs** [comment]
@faddat Thanks for lending a hand here! 

I hadn't taken a stab at the integration tests yet - I suspect those will need quite a bit of refactoring to get them to work. I can give it a pass tomorrow. Thanks for removing brew - that was long overdue!

Can you confirm you able to at least start dockernet without an issue?

**Comment by sampocs** [comment]
@nghuyenthevinh2000 Actually disregard that - I don't think the relayer update resolved it. I'll continue to debug

File: Dockerfile.gaia [hunk 1]
```diff
@@ -15,7 +15,7 @@ FROM alpine:3.15
 COPY --from=builder /go/bin/gaiad /usr/local/bin/
 RUN adduser -S -h /home/gaia -D gaia -u 1000
 
-USER gaia
+USER 1000
```
Comment (vish-stride) [inline Dockerfile.gaia:18]: interesting that this is platform dependent

File: scripts/build.sh [hunk 2]
```diff
@@ -14,16 +15,27 @@ build_local_and_docker() {
    printf '%s' "Building $title Locally...  "
    cwd=$PWD
    cd $folder
-   go build -mod=readonly -trimpath -o $BUILDDIR ./... 2>&1 | grep -v -E "deprecated|keychain";
-   local_build_succeeded=$?
+   GOBIN=$BUILDDIR go install -mod=readonly -trimpath ./... 2>&1 | grep -v -E "deprecated|keychain" | true
```
Comment (vish-stride) [inline scripts/build.sh:18]: surprised `build` vs `install` is so different!

File: scripts/register_host.sh [hunk 3]
```diff
@@ -20,7 +20,7 @@ echo "$CHAIN_ID - Registering host zone..."
 $STRIDE_MAIN_CMD tx stakeibc register-host-zone \
     $CONNECTION $HOST_DENOM $ADDRESS_PREFIX $IBC_DENOM $CHANNEL 1 \
     --gas 1000000 --from $STRIDE_ADMIN_ACCT --home $SCRIPT_DIR/state/stride1 -y | TRIM_TX
-sleep 4
+sleep 10
```
Comment (vish-stride) [inline scripts/register_host.sh:23]: slow machines?

File: scripts/register_host.sh [hunk 4]
```diff
@@ -30,12 +30,21 @@ for (( i=1; i <= $NUM_VALS; i++ )); do
...
 done
 
+timeout=100
 while true; do
     if ! $STRIDE_MAIN_CMD q stakeibc list-host-zone | grep Account | grep -q null; then
-        sleep 1
         break
+    else
+        if [[ "$timeout" == "0" ]]; then 
+            echo "ERROR - Unable to register host zones."
```
Comment (vish-stride) [inline scripts/register_host.sh:42]: Nice this is a great change!

**Review by vish-stride — APPROVED** [review]
Looks great, very nice that we're finally able to run consistently on linux!

File: scripts/register_host.sh (hunk 3 above)
Comment (nghuyenthevinh2000) [reply scripts/register_host.sh:23]: yep

File: scripts/build.sh (hunk 2 above)
Comment (nghuyenthevinh2000) [reply scripts/build.sh:18]: go build is inconsistent

File: scripts/register_host.sh (hunk 3 above)
Comment (sampocs) [reply scripts/register_host.sh:23]: yup!

**Comment by sampocs** [comment]
Merging this in now as there are various bug fixes and it works across the majority of tested machines.
Will tackle integration tests in a separate PR as well as any other outstanding issues.

**Pull request merged**